src/prommis/solvent_extraction/tests/test_sx_model_dynamic.py

```diff
@@ -0,0 +1,261 @@
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Y"].fix(5.67e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["La"].fix(1.78e-05)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Ce"].fix(4.019e-5)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Pr"].fix(6.73e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Nd"].fix(1.82e-5)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Sm"].fix(3.285e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Gd"].fix(1.55e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Dy"].fix(9e-7)
+
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Al"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ca"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Fe"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sc"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Y"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["La"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ce"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Pr"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Nd"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sm"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Gd"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Dy"].fix(1e-9)
+
+        m.fs.solex.mscontactor.organic_inlet_state[:].flow_vol.fix(62.01)
+
+        m.fs.solex.mscontactor.organic[0, :].flow_vol.fix(62.01)
+
+        return m
+
+    @pytest.mark.component
+    def test_structural_issues(self, SolEx_frame):
+        model = SolEx_frame
+        dt = DiagnosticsToolbox(model)
+        dt.assert_no_structural_warnings()
+
+    @pytest.mark.component
+    def test_block_triangularization(self, SolEx_frame):
+        model = SolEx_frame
+        initializer = BlockTriangularizationInitializer(constraint_tolerance=2e-5)
+        initializer.initialize(model.fs.solex)
+
+        assert initializer.summary[model.fs.solex]["status"] == InitializationStatus.Ok
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, SolEx_frame):
+        m = SolEx_frame
+        results = solver.solve(m)
+
+        # Check for optimal solution
+        assert check_optimal_termination(results)
+
+    @pytest.mark.integration
+    def test_solution(self, SolEx_frame):
+        m = SolEx_frame
+        time_duration = 60
+        assert value(
+            m.fs.solex.mscontactor.aqueous[time_duration, 3].conc_mass_comp["HSO4"]
+        ) == pytest.approx(7.651e-5, rel=1e-1)
```

Comment:
This tolerance is way too loose - you are allowing a 10% variation in the value here. This should be rel=1e-4 at the most, and generally more like 1e-5.

Reply:
I have given rel 1e-4 and an abs value of 1e-10

Reply:
Did you try `rel=1e-5`? 1e-4 was my upper bound on the tolerance, and you should generally be able to get 1e-5 without issue.

Second, an `abs` tolerance of 1e-10 is way to tight to be meaningful (unless you have scaling) - solver tolerance is only 1e-8, so anything smaller than that will be lost to noise. Also, you generally only need to provide one of these (this is a test so you know the expected result and can determine a priori which tolerance will be the loosest).

Reply:
yes I tried, but there was assertion error after the 4th decimal place